app/com/kashoo/ws/RequestRateLimits.scala

```diff
@@ -41,13 +44,13 @@ object RequestRateLimits {
 
   def apply(rrl: RequestRateLimit*): RequestRateLimits = Set[RequestRateLimit](rrl:_*)
 
-  def apply(config: Configuration): RequestRateLimits = {
-    val rateLimiterEc = config.getConfig("ws.limited.execution-context") match {
-      case Some(ecConfig) => Akka.system.dispatchers.lookup("ws.limited.execution-context")
-      case None => play.api.libs.concurrent.Execution.Implicits.defaultContext
+  def apply(config: Configuration, actorSystem: ActorSystem)(implicit ec: ExecutionContext): RequestRateLimits = {
```

Comment:
it's not uncommon to have `actorSystem` be an implicit parameter, similar to `ec: ExecutionContext`